src/uct/tcp/tcp_ep.c

```diff
@@ -535,14 +535,13 @@ static inline unsigned uct_tcp_ep_recv(uct_tcp_ep_t *ep, size_t *recv_length)
         } else {
             uct_tcp_ep_handle_disconnected(ep, &ep->rx);
         }
-        *recv_length = 0;
         return 0;
     }
 
-    ep->rx.length += *recv_length;
-    ucs_trace_data("tcp_ep %p: recvd %zu bytes", ep, *recv_length);
+    ep->rx.length += recv_length;
+    ucs_trace_data("tcp_ep %p: recvd %zu bytes", ep, recv_length);
 
-    return 1;
+    return recv_length > 0;
```

Comment:
need to assert that length > 0
```
ucs_socket_do_io_nb()
...
    if (ret == 0) {
        ucs_trace("fd %d is closed", fd);
        return UCS_ERR_CANCELED; /* Connection closed */
    }
```

Reply:
good catch, then we just can return `1`
done